### ISSUE TITLE
Ignore list?

### ISSUE THREAD

**Issue by vuxyp** [ghacksuserjs/ghacks-user.js#784]
Is there any possibility to add feature like ignore-list which is implemented on pyllyukko's user-js?

https://github.com/pyllyukko/user.js/blob/master/ignore.list

Thank you

**Comment by Thorin-Oakenpants** [comment]
What's it used for, by who, and how? How do they "comparison with Firefox upstream prefs". This is outside the scope of this project as an end product for the user, IMO.

Just explain to me why and what this is for

**Comment by vuxyp** [comment]
@Thorin-Oakenpants ignore-list is the list which is ignored if the user.js will activate it. 

A very simple example:

If I disable webgl using:

> webgl.disabled=true

I don't need other webgl privacy preferences. So, as far as I know, "pyllyukko" does not puts all unnecessary preferences on user js file. So the userjs file is clean.

If you read the file (the same file on my post):
https://github.com/pyllyukko/user.js/blob/master/ignore.list

you can see the notes of each block. it writes why it is unnecessary to add that block to user.js.

**Comment by Thorin-Oakenpants** [comment]
I don't understand. ELI5 .. I come in here, I get the user.js, I add some overrides to it, and I use it.

What do I do with an ignore.list?

**Comment by vuxyp** [comment]
It's simple :)

If you have on your user.js file:

> webgl.disabled = true

You have on your ignore-list.js:

```
"webgl.allow-fb-invalidation"
"webgl.angle.force-d3d11"
"webgl.angle.force-warp"
"webgl.angle.try-d3d11"
"webgl.bypass-shader-validation"

... (See the full list here: https://github.com/pyllyukko/user.js/blob/master/ignore.list#L341)
```

Why ?
"webgl.allow-fb-invalidation" is unnecessary.

**Comment by vuxyp** [comment]
Another clean and simple example:

https://github.com/pyllyukko/user.js/blob/master/ignore.list#L371

**Comment by vuxyp** [comment]
This feature is not a must, but:
- it makes more clean the user js file.
- if the want enable only one feature (for example I want to enable **only** webgl on my profile), I can only change one preference "webgl.diasabled".

**Comment by Thorin-Oakenpants** [comment]
OK, I'll try **one more time** ....

**HOW** do you use it? Not how does it work (I can gather enough from the fact that it says `ignore` and you're using two files: ignore.list and user.js).

**Comment by Thorin-Oakenpants** [comment]
nevermind

> So, as far as I know, "pyllyukko" does not puts all unnecessary preferences on user js file. So the userjs file is clean

Our file is already clean. The user does not need to pull in any data from Firefox "upstream" and get things excluded. pyllyukko doesn't build his user.js this way - it's been crafted manually - just like ours. We have no need for an "ignore" list - @earthlng does this for us with his own tools, and a different method by looking at diffs between each stable release: and that's the same sort of thing pyllyukko was doing: his ignore list is things he simply wanted to remove from *his results* to save time (not from the *user.js*)
